yt_dlp/extractor/taptap.py

```diff
@@ -0,0 +1,275 @@
+import re
+import uuid
+
+from .common import InfoExtractor
+from ..utils import (
+    clean_html,
+    int_or_none,
+    join_nonempty,
+    str_or_none,
+    url_or_none,
+)
+from ..utils.traversal import traverse_obj
+
+
+class TapTapBaseIE(InfoExtractor):
+    _X_UA = 'V=1&PN=WebApp&LANG=zh_CN&VN_CODE=102&LOC=CN&PLT=PC&DS=Android&UID={uuid}&OS=Windows&OSV=10&DT=PC'
+    _VIDEO_API = 'https://www.taptap.cn/webapiv2/video-resource/v1/multi-get'
+    _INFO_API = None
+    _INFO_QUERY_KEY = 'id'
+    _DATA_PATH = None
+    _ID_PATH = None
+    _META_PATH = None
+
+    def _get_api(self, url, video_id, query, **kwargs):
+        query = {**query, 'X-UA': self._X_UA.format(uuid=uuid.uuid4())}
+        return self._download_json(url, video_id, query=query, **kwargs)['data']
+
+    def _extract_video(self, video_id):
+        video_data = self._get_api(self._VIDEO_API, video_id, query={'video_ids': video_id})['list'][0]
+
+        # h265 playlist contains both h265 and h264 formats
+        video_url = traverse_obj(video_data, ('play_url', ('url_h265', 'url'), {url_or_none}))[0]
```

Comment:
```suggestion
        video_url = traverse_obj(video_data, ('play_url', ('url_h265', 'url'), {url_or_none}, any))
```
That way the `None` will cause `self._extract_m3u8_formats` to fail gracefully